Private Events web application with user authentication using Rails Associations

emasdev opened the pull request:
## About Private Events
- Users can create events and sign to attendee an event. Events take place at a specific date and at a location.

## Goal
- The main goal is to demonstrate mastering in building Rails Associations

## Features

If you're not an authenticated user

- Create User
- Log in

If you are an authenticated user

- See the list of Events available divided by upcoming and previous events
- See users attendance list
- See the attendance list for a specific event
- Register to attend an event

Rspec Testing available

emasdev committed:
Start rails project, user model and controller generated

MarilenaRoque committed:
Build controllers and views for user Model

- View/Controller for show method
- View/Controler for new method
- View for Log In

emasdev committed:
Add Login/Logout for Users using sessions

MarilenaRoque committed:
Set Post Model

- Generate the model/controllers/view for Posts
- Set the relation between Post and User
- Set event creation by the user

MarilenaRoque committed:
Add user index to post model

emasdev committed:
Add Events with User associations

- Add event views
- Add event/user associations in Models
- Add event controller

MarilenaRoque committed:
Build Table to associate Attendances

- Create model with attributes
- Set the realtions

emasdev committed:
Fix foreign key on Event & User model

emasdev committed:
Add event_attending functionality and change user_id for creator_id in Events

-Change user_id for creator_id
-Use creator alias for event user creator
-Created event_attending controller
-Add attende button to events list

MarilenaRoque committed:
Add the create method to build Attendances

Built nested resources
Set create methods

emasdev committed:
Show list of attendance events and attendees

-List attended events in user show view
-List attendees of an event in event show view

emasdev committed:
update bootstrap

MarilenaRoque committed:
Added Scopes to separate events by date

Set Events
Set view display for users

emasdev committed:
Modify the Event Index page to list all events, update UI/UX

MarilenaRoque committed:
Tests and README.md informations

Add stickler and rubocop files
Add test gems as dependencies
Add README.md information

emasdev committed:
Add user model rspec test

MarilenaRoque committed:
Add presence required to user name

MarilenaRoque committed:
Set tests and attendance button

- Set image for events attended
- Set button for events not attended
- Set association tests

emasdev committed:
Restrict creation of events if user is not login

emasdev committed:
Test/Readme/Rubocop fixes

-Rspec test user/event/event_attending
-Update Readme file
-Fix rubocop offenses

maelfosso commented on db/schema.rb:
```diff
@@ -0,0 +1,39 @@
+    t.integer "attended_event_id"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
```
Please, add index for the referential fields.

maelfosso commented on app/views/layouts/application.html.erb:
```diff
@@ -0,0 +1,50 @@
+        <% else %>
+          <%= link_to "Login", '/login', method: :get, :class => 'navbar-link' %> |
+          <%= link_to "Sign Up", '/users/new', method: :get, :class => 'navbar-link' %>
+        <%end%>
```
Remove logic from views please. implement them in view helpers.

maelfosso commented on app/controllers/events_controller.rb:
```diff
@@ -0,0 +1,31 @@
+  def new
+    if current_user.nil?
+      redirect_to '/login'
+    else
```
No needed. Add the new method into the `before_action` statement

maelfosso reviewed (changes requested):
# Changes required :red_circle: 

Hello,
Thank you for the job done so far. However, there are some missing requirements

1. Kindly add integration tests. It required. [Read it here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png)

2 Read the comments left below.

Kindly make the suggested changes  and submit a new code review request
Thanks

Mael FOSSO

MarilenaRoque committed:
Build Tests for Users view

-Testing index page
-Testing users creation
-Testing buttons

emasdev committed:
Add integration testing for events and attending events

MarilenaRoque committed:
Removing logic from the view

Build a helper to render a partial
Add partials code

emasdev committed:
Add indexes to EventAttending
Fix Rubocop offenses
Change helper method before_action to redirect to Login when not signed in

MarilenaRoque commented:
Hello @maelfosso. Thanks for the review, the correction, and the clarifications about the requirements. :+1: 
We worked in all the topics listed and all the changes were implemented.

- [x] Removed all the logic inside the view using helpers
- [x] Added index for all the foreign keys
- [x] Built integration tests
- [x] Removed the if for the redirection

# :rocket:

mosesogwo reviewed (approved):
**Hello!:wave: Good job with the changes :+1:**

You can now merge this PR.

_**Cheers:stars:**_

### Status: Approved :heavy_check_mark:

MarilenaRoque commented:
# Thank you @mosesogwo! :rocket:
